library/std/src/sys/unix/fd.rs

```diff
@@ -8,9 +8,82 @@ use crate::sys_common::AsInner;
 
 use libc::{c_int, c_void};
 
+// Android's libc allows for file descriptors to be marked with a tag that marks ownership.
+// If a tagged file descriptor is closed with the wrong tag, either a backtrace is printed to logs,
+// or the process aborts, depending on process configuration.
+#[cfg(target_os = "android")]
+mod android {
+    use crate::sync::atomic::{AtomicU64, Ordering};
+    use libc::c_int;
+
+    #[derive(Debug)]
+    pub struct CloseTag(u64);
+
+    static FD_TAG: AtomicU64 = AtomicU64::new(0);
+
+    fn next_tag() -> u64 {
```

Comment:
This could use a comment that it will never return 0 which is good to know since 0 has special meaning.